pinot-core/src/main/java/com/linkedin/pinot/core/query/executor/ServerQueryExecutorV1Impl.java

```diff
@@ -221,6 +233,46 @@ private long pruneSegments(TableDataManager tableDataManager, List<SegmentDataMa
     return totalRawDocs;
   }
 
+  /**
+   * Helper method to compute missing segments
+   *
+   * @param segmentsToQuery segments that need to be queried according to server query request
+   * @param segmentDataManagers acquired segment data managers from table data manager
+   * @return a list of missing segments
+   */
+  @Nonnull
```

Comment:
I think the new convention is to not use Nonnull annotation

Reply:
removed the annotation